functional-tests/test-bed/src/main/java/org/arquillian/smart/testing/ftest/testbed/project/Project.java

```diff
@@ -36,6 +42,32 @@ public ProjectConfigurator configureSmartTesting() {
         return new ChangeApplier(repository).applyAsCommits(changeDescriptions);
     }
 
+    Project checkoutBranch(String branchName) {
+        try {
+            git.checkout().setName(branchName).call();
```

Comment:
Well in original issue after discussing we agreed to do using a branch for this reason I used this approach, but I agree that maybe using the commit thing might be better at least to have something more similar